pages/index.js

```diff
@@ -0,0 +1,60 @@
+import envFlag from 'node-env-flag';
+import React from 'react';
+import Meta from '../frontend/components/meta';
+import Header from '../frontend/components/header';
+import SuggestionAndSearch from '../frontend/components/suggestion-and-search';
+import { BadgeExamples } from '../frontend/components/badge-examples';
+import MarkupModal from '../frontend/components/markup-modal';
+import Usage from '../frontend/components/usage';
+import Footer from '../frontend/components/footer';
+import badgeExampleData from '../lib/all-badge-examples';
+import filterExamples from '../frontend/lib/filter-examples';
+
+const baseUri = process.env.BASE_URL || 'http://localhost:8080';
+const isProductionBuild = envFlag(process.env.PRODUCTION_BUILD, false);
+
+export default class IndexPage extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = { query: null, example: null };
+  }
+
+  render() {
+    // This approach is the slightest bit slow. Since all the badges are on
+    // the screen at the beginning, we might get a more responsive search by
+    // adjusting visibility of the elements rather than removing them from the
+    // DOM and recreating them, as this does now. That's what the original
+    // code did.
+    const filteredExamples = filterExamples(badgeExampleData, this.state.query);
+
+    return (
+      <div>
+        <Meta />
+        <Header />
+        <MarkupModal
+          example={this.state.example}
+          onRequestClose={() => { this.setState({ example: null }); }}
+          baseUri={baseUri} />
+        <section>
+          <SuggestionAndSearch
+            queryChanged={query => { this.setState({ query }); }}
+            onBadgeClick={example => { this.setState({ example }); }}
+            baseUri={baseUri}
+            isProductionBuild={isProductionBuild} />
+          <a
+            href="https://gratipay.com/Shields/"
+            style={{ textDecoration: 'none', color: 'rgba(0,0,0,0.1)' }}>
```

Comment:
Use CSS file for this? CSS files are better for perf. If you do want to switch to using "CSS in JS", consider using a library like Glamor.